dependencies.xml

```diff
@@ -64,13 +64,11 @@ Note all install methods after "main" take
     <numexpr os='linux'/>
     <cmake skip_check='True' optional='True'/>
     <dask source="pip" pip_extra="[complete]"/>
-    <ray source="pip" pip_extra="[default]" os='mac,linux'>2.2</ray>
-    <ray source="pip" pip_extra="[default]" os='windows'>1.13</ray>
-    <pydantic source="pip" os='mac,linux'>1</pydantic> <!-- ray 2.2 can't use pydantic 2 -->
+    <ray source="pip" pip_extra="[default]">2.6</ray>
     <!-- redis is needed by ray, but on windows, this seems to need to be explicitly stated -->
     <redis source="pip" os='windows'/>
     <imageio source="pip">2.22</imageio>
-    <smt/>
+    <smt optional='True'/> <!-- not available on macos arm64 -->
```

Comment:
The code changes are good to me. However I do have a concern about BayesianOptimization tests. I guess we have only one machine that is forced to install all optional libraries. In this case, we will not test the Bayesian Optimization in other machines. I would suggest to at least enable optional libraries installations for MAC, Windows and Sawtooth. @joshua-cogliati-inl Please let me know what do you think.

Reply:
Hm, installing optional libraries will slow down things by a bit.

Reply:
Maybe we should add an `machine='x86_64'` or similar that only installs based on what `platform.machine()` returns.

Reply:
I like the idea, either add another attribute for dependencies or extend optional to accept values that indicate machine info.

Reply:
I added an attribute for the dependencies.